src/utils/fee_calculator.ts

```diff
@@ -11,33 +12,48 @@ import {
     VolumeBasedFeeConfig,
 } from '../types';
 
-/**
- * `transferFrom` estimated gas:
- * - Decrease balance of the owner (SLOAD + SSTORE): 24,000
- * - Increase balance of the spender (SLOAD + SSTORE): 24,000
- * - Update allowance of the spender (SLOAD + SSTORE): 24,000
- */
-const TRANSFER_FROM_GAS = new BigNumber(72e3);
+interface OnChainTransfer {
+    feeToken: string;
+    feeAmount: BigNumber;
+    feeRecipient: string;
+    affilateFeeType: AffiliateFeeType;
+}
```

Comment:
Hmmm... is this the same as the `AffiliateFeeAmount` type? I recommend against reusing it (especially the concept of `AffiliateFeeType`). Instead, I would just have the following fields and handle conversion outside of this file


```suggestion
interface OnChainTransfer {
    feeToken: string;
    feeAmount: BigNumber;
    feeRecipient: string;
}
```